### PR TITLE
[Swift Package Manager] Test removing the last Flutter plugin

### PR DESCRIPTION
**Pull request by loic-sharma**
The Flutter tool has a bug where removing the last Flutter plugin does not correctly update the CocoaPods integration.

This adds a test to ensure  that the generated Swift package is properly updated when the last Flutter plugin is removed.

See: https://github.com/flutter/flutter/issues/11819#issuecomment-2289782626

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter_tools/test/integration.shard/swift_package_manager_test.dart [hunk 1]
```diff
@@ -607,4 +607,87 @@ void main() {
...
+        appDirectoryPath,
+        options: <String>['ios', '--config-only', '-v'],
+      );
+
+      // Verify the generated Swift package does not depend on the plugin.
+      expect(generatedManifestFile.existsSync(), isTrue);
+
+      generatedManifest = generatedManifestFile.readAsStringSync();
+
+      expect(generatedManifest.contains(generatedSwiftDependency), isFalse);
```
Comment (jmagman) [inline packages/flutter_tools/test/integration.shard/swift_package_manager_test.dart:684]: What's in the Package.swift when the last dependency is removed? Is it empty?

Comment (loic-sharma) [reply packages/flutter_tools/test/integration.shard/swift_package_manager_test.dart:684]: It adds an effectively empty local package with an empty target:

```swift
// swift-tools-version: 5.9
// The swift-tools-version declares the minimum version of Swift required to build this package.
//
//  Generated file. Do not edit.
//

import PackageDescription

let package = Package(
    name: "FlutterGeneratedPluginSwiftPackage",
    platforms: [
        .iOS("12.0")
    ],
    products: [
        .library(name: "FlutterGeneratedPluginSwiftPackage", type: .static, targets: ["FlutterGeneratedPluginSwiftPackage"])
    ],
    dependencies: [
        
    ],
    targets: [
        .target(
            name: "FlutterGeneratedPluginSwiftPackage"
        )
    ]
)

```

Comment (loic-sharma) [reply packages/flutter_tools/test/integration.shard/swift_package_manager_test.dart:684]: Now that I think about this, let me update the assertion to check for this:

```
    dependencies: [
        
    ],
```

Comment (loic-sharma) [reply packages/flutter_tools/test/integration.shard/swift_package_manager_test.dart:685]: Updated!

File: packages/flutter_tools/test/integration.shard/swift_package_manager_test.dart [hunk 2]
```diff
@@ -607,4 +607,89 @@ void main() {
...
+        flutterBin,
+        appDirectoryPath,
+        options: <String>['ios', '--config-only', '-v'],
+      );
+
+      // Verify the generated Swift package does not depend on the plugin.
+      expect(generatedManifestFile.existsSync(), isTrue);
+
+      generatedManifest = generatedManifestFile.readAsStringSync();
+      const String emptyDependencies = 'dependencies: [\n        \n    ],\n';
```
Comment (loic-sharma) [inline packages/flutter_tools/test/integration.shard/swift_package_manager_test.dart:683]: This doesn't use the multi-line string syntax as code editors often aggressively trim trailing whitespace on save, which breaks the test.

**Review by jmagman — APPROVED** [review]
LGTM, thanks for adding this test!

**Pull request merged**